subsys/net/lib/lwm2m/lwm2m_obj_security.c

```diff
@@ -138,6 +169,13 @@ static struct lwm2m_engine_obj_inst *security_create(uint16_t obj_inst_id)
 	INIT_OBJ_RES_DATA(SECURITY_SHORT_SERVER_ID, res[index], i,
 			  res_inst[index], j,
 			  &short_server_id[index], sizeof(*short_server_id));
+#if defined(CONFIG_LWM2M_SECURITY_OBJECT_VERSION_1_1)
+	INIT_OBJ_RES_OPTDATA(SECURITY_MATCHING_TYPE_ID, res[index], i, res_inst[index], j);
+	INIT_OBJ_RES_OPTDATA(SECURITY_SNI_ID, res[index], i, res_inst[index], j);
+	INIT_OBJ_RES_OPTDATA(SECURITY_CERTIFICATE_USAGE_ID, res[index], i, res_inst[index], j);
+	INIT_OBJ_RES_OPTDATA(SECURITY_DTLS_TLS_CIPHERSUITE_ID, res[index], i, res_inst[index], j);
```

Comment:
I think this should use `INIT_OBJ_RES_MULTI_OPTDATA` since it's a multi-instance resource.